lib/parsers/src/reasoning/gpt_oss_parser.rs

```diff
@@ -56,12 +56,28 @@ impl GptOssReasoningParser {
     }
 }
 
+fn encode_text_to_tokens(text: &str) -> anyhow::Result<Vec<u32>> {
+    let enc = get_harmony_encoding()
+        .as_ref()
+        .map_err(|e| anyhow::anyhow!("Failed to get harmony encoding: {e}"))?;
+    Ok(enc.tokenizer().encode_with_special_tokens(text))
+}
+
 impl ReasoningParser for GptOssReasoningParser {
-    fn detect_and_parse_reasoning(&mut self, _text: &str, token_ids: &[u32]) -> ParserResult {
-        tracing::debug!(
-            "detect_and_parse_reasoning called with {} token_ids",
-            token_ids.len()
-        );
+    fn detect_and_parse_reasoning(&mut self, text: &str, token_ids: &[u32]) -> ParserResult {
+        let token_ids = if token_ids.is_empty() {
+            // WAR: Since we are moving to just text based reasoning parsing, converting to token_ids now using harmony encoding
+            let encoded_tokens = match encode_text_to_tokens(text) {
```

Comment:
CC @biswapanda for viz - it would be helpful to see if there is a noticeable performance impact on the GPT-OSS NVL72 1.5-2M tokens/sec recipe before and after this PR.